app/components/curriculum-inventory/sequence-block-list.js

```diff
@@ -1,109 +1,39 @@
-import Component from '@ember/component';
-import { computed } from '@ember/object';
-import ObjectProxy from '@ember/object/proxy';
-import { isPresent } from '@ember/utils';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { dropTask } from 'ember-concurrency';
 import { all } from 'rsvp';
 
-const SequenceBlockProxy = ObjectProxy.extend({
-  content: null,
-  showRemoveConfirmation: false,
-});
-
-export default Component.extend({
-  tagName: '',
-  canUpdate: false,
-  editorOn: false,
-  isSaving: null,
-  parent: null,
-  report: null,
-  saved: false,
-  savedBlock: null,
-  sequenceBlocks: null,
-
-  isInOrderedSequence: computed('parent', function () {
-    const parent = this.parent;
-    return isPresent(parent) && parent.get('isOrdered');
-  }),
-
-  sortedBlocks: computed(
-    'sequenceBlocks.@each.orderInSequence',
-    'parent.childSequenceOrder',
-    async function () {
-      const parent = this.parent;
-      const sequenceBlocks = this.sequenceBlocks;
-
-      if (isPresent(parent) && parent.isOrdered) {
-        return sequenceBlocks
-          .sortBy('orderInSequence', 'title', 'id')
-          .map((block) => SequenceBlockProxy.create({ content: block }));
-      }
-
-      if (!sequenceBlocks.length) {
-        return sequenceBlocks;
-      }
-
-      const blockProxies = await all(
-        sequenceBlocks.map(async (block) => {
-          const proxy = ObjectProxy.create({ content: block, level: null });
-          const academicLevel = await block.academicLevel;
-          proxy.set('level', academicLevel.level);
-          return proxy;
-        })
-      );
-      return blockProxies.sortBy('level', 'startDate', 'title', 'id').map((sortedProxy) => {
-        return SequenceBlockProxy.create({
-          content: sortedProxy.content,
-        });
-      });
+export default class SequenceBlockListComponent extends Component {
+  @tracked editorOn = false;
+  @tracked savedBlock;
+
+  get isInOrderedSequence() {
+    return this.args.parent && this.args.parent.isOrdered;
+  }
+
+  @action
+  toggleEditor() {
+    this.editorOn = !this.editorOn;
+    this.savedBlock = null;
+  }
+
+  @action
+  cancel() {
+    this.editorOn = false;
+  }
+
+  @dropTask
+  *save(block) {
+    this.editorOn = false;
+    this.savedBlock = yield block.save();
+    // adding/updating a sequence block will have side-effects on its siblings if the given block is nested
+    // inside an "ordered" sequence block. they all get re-sorted server-side.
+    // therefore, we must reload them here in order to get those updated sort order values.
+    // [ST 2021/03/16]
+    if (this.args.parent) {
+      const blocks = yield this.args.parent.children;
+      yield all(blocks.invoke('reload'));
```

Comment:
This will load them one at a time in a bunch of requests, which is maybe ok depending on how many there are. Could also leverage JSON:API side loading to get all children in the same request with:

```
this.store.findRecord('curriculum_inventory_sequence_block', this.args.parent.id, {
   include: 'children',
   reload: true,
});
```

Reply:
:heavy_check_mark: